src/Core/AdminConsole/Services/Implementations/OrganizationService.cs

```diff
@@ -1176,20 +1181,41 @@ await _referenceEventService.RaiseEventAsync(
             var currentOrganization = await _organizationRepository.GetByIdAsync(organization.Id);
 
             // Revert autoscaling
-            if (initialSeatCount.HasValue && currentOrganization.Seats.HasValue && currentOrganization.Seats.Value != initialSeatCount.Value)
-            {
-                await AdjustSeatsAsync(organization, initialSeatCount.Value - currentOrganization.Seats.Value);
-            }
+            var seatAdjustment = initialSeatCount.HasValue && currentOrganization.Seats.HasValue
+                ? initialSeatCount.Value - currentOrganization.Seats.Value
+                : 0;
+
+            var smSeatAdjustment = initialSmSeatCount.HasValue && currentOrganization.SmSeats.HasValue
+                ? initialSmSeatCount.Value - currentOrganization.SmSeats.Value
+                : 0;
 
-            // Revert SmSeat autoscaling
-            if (initialSmSeatCount.HasValue && currentOrganization.SmSeats.HasValue &&
-                currentOrganization.SmSeats.Value != initialSmSeatCount.Value)
+            // There is a possibility that the initial state may have a lower PM seat count than SM seat count.
+            // In this case we need to revert in the opposite order
+            if (seatAdjustment != 0 && smSeatAdjustment != 0 && initialSeatCount.Value > currentOrganization.SmSeats.Value)
```

Comment:
I don't think there's any need for this conditional logic, unless you found an issue in your testing.

I just meant swap the current code blocks, i.e.

```cs
            // Revert SmSeat autoscaling
            // Do this first so that SmSeats never exceed PM seats (due to current billing requirements)
            if (initialSmSeatCount.HasValue && currentOrganization.SmSeats.HasValue &&
                currentOrganization.SmSeats.Value != initialSmSeatCount.Value)
            {
                var smSubscriptionUpdateRevert = new SecretsManagerSubscriptionUpdate(currentOrganization, false)
                {
                    SmSeats = initialSmSeatCount.Value
                };
                await _updateSecretsManagerSubscriptionCommand.UpdateSubscriptionAsync(smSubscriptionUpdateRevert);
            }

            // Revert PM autoscaling
            if (initialSeatCount.HasValue && currentOrganization.Seats.HasValue && currentOrganization.Seats.Value != initialSeatCount.Value)
            {
                await AdjustSeatsAsync(organization, initialSeatCount.Value - currentOrganization.Seats.Value);
            }
```